frictionless/plugins/ckan.py

```diff
@@ -230,32 +223,61 @@ def write_table_convert_field_type(self, type):
             "object": "json",
             "array": "text[]",
             "geojson": "json",
-            "date": "date",
+            "date": "text",
             "time": "time",
             "year": "int",
             "datetime": "timestamp",
         }
-        try:
-            return DESCRIPTOR_TYPE_MAPPING[type]
-        except KeyError:
-            log.warn("Unsupported descriptor type '{}'.".format(type))
-            return None
+        return DESCRIPTOR_TYPE_MAPPING.get(type, "text")
 
-    def write_table_row_stream(self, name, row_stream, method="upsert"):
-        table = self.read_table(name)
+    def _write_table_row_stream(self, name, row_stream):
+        table = self._read_table(name)
         datastore_upsert_url = "{}/datastore_upsert".format(self.__base_endpoint)
-        records = []
-        for r in row_stream:
-            #  r = self.__mapper.convert_row(r, table.eschema)
-            records.append(r)
+        records = [r.to_dict(json=True) for r in row_stream]
         params = {
             "resource_id": table.name,
-            "method": method,
+            "method": "insert",
             "force": True,
             "records": records,
         }
         self.__make_ckan_request(datastore_upsert_url, method="POST", json=params)
 
+    def write_resource(self, resource, *, force=False):
+        self._write_table(resource, force=force)
+        self._write_table_row_stream(resource.name, resource.read_row_stream())
+
+    def write_package(self, package, *, force=False):
+        for resource in package.resources:
+            self.write_resource(resource, force=force)
+
+    # Delete
+
+    def _write_table_remove(self, name, ignore=False):
+        # Check existent
+        if name not in self._read_table_names():
+            if not ignore:
+                note = f'Table "{name}" does not exist'
+                raise exceptions.FrictionlessException(errors.StorageError(note=note))
+
+        # Remove from ckan
+        datastore_delete_url = "{}/datastore_delete".format(self.__base_endpoint)
+        params = {"resource_id": name, "force": True}
+        self.__make_ckan_request(datastore_delete_url, method="POST", json=params)
+
+        # Remove from table
+        if name in self.__tables:
+            del self.__tables[name]
+
+        # Invalidate cache
+        self.__bucket_cache = None
+
+    def delete_resource(self, name, *, ignore=False):
+        self._write_table_remove(name, ignore=ignore)
+
+    def delete_package(self, names, *, ignore=False):
```

Comment:
Bit of a tangent, but I find it really surprising that `write_package()` takes a `Package` object but `delete_package()` takes a list of resource names.
Any reason for that design decision? Would it make sense to change the storage API so `delete_package()` takes a `Package` object ahead of the 4.0 release?